tests/Integration/Behaviour/Features/Context/Domain/OrderReturnFeatureContext.php

```diff
@@ -0,0 +1,128 @@
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace Tests\Integration\Behaviour\Features\Context\Domain;
+
+use Configuration;
+use Order;
+use OrderReturn;
+use OrderReturnState;
+use PrestaShop\PrestaShop\Core\Domain\OrderReturn\Command\UpdateOrderReturnStateCommand;
+use PrestaShop\PrestaShop\Core\Domain\OrderReturn\Exception\OrderReturnConstraintException;
+use PrestaShopDatabaseException;
+use PrestaShopException;
+use RuntimeException;
+use Tests\Integration\Behaviour\Features\Context\SharedStorage;
+
+class OrderReturnFeatureContext extends AbstractDomainFeatureContext
+{
+    /**
+     * @When I add order return state :orderReturnStateReference
+     *
+     * @param string $orderReturnStateReference
+     *
+     * @throws \PrestaShopDatabaseException
+     * @throws \PrestaShopException
+     */
+    public function createOrderReturnState(string $orderReturnStateReference): void
+    {
+        $orderReturnState = new OrderReturnState(null, Configuration::get('PS_LANG_DEFAULT'));
+        $orderReturnState->name = 'New order return state';
+        $orderReturnState->add();
+        SharedStorage::getStorage()->set($orderReturnStateReference, $orderReturnState->id);
+    }
+
+    /**
+     * @When I add order return :orderReturnReference from order :orderReference
+     *
+     * @param string $orderReturnReference
+     * @param string $orderReference
+     *
+     * @throws PrestaShopDatabaseException
+     * @throws PrestaShopException
+     */
+    public function createOrderReturn(string $orderReturnReference, string $orderReference): void
+    {
+        $orderId = SharedStorage::getStorage()->get($orderReference);
+        $order = new Order($orderId);
+        $orderReturn = new OrderReturn();
+        $orderReturn->id_customer = (int) $order->id_customer;
+        $orderReturn->id_order = $order->id;
+        $orderReturn->question = 'Why?';
+        $orderReturn->state = 1;
```

Comment:
This refers to an `OrderState` in database right? Maybe you should rather use a state you created via a reference stored in the SharedStorage?

It's just that the hardcode 1 value seems a bit arbitrary, as we don't even know which state it relates to or if it even exists

Reply:
I have state as 1 because first state doesn't really matter.

What I am testing for is that it correctly changes to state referenced in storage after update.